crates/apollo-router-core/src/schema.rs

```diff
@@ -117,11 +117,21 @@ impl std::str::FromStr for Schema {
 
                                                     let arg_value =
                                                         argument.value().and_then(|s| {
-                                                            s.to_string()
-                                                                .trim_end()
-                                                                .strip_prefix('"')
-                                                                .and_then(|s| s.strip_suffix('"'))
-                                                                .map(|s| s.to_owned())
+                                                            // This is a temporary workaround until we have nice semantic analysis
+                                                            s.syntax()
+                                                                .green()
+                                                                .children()
+                                                                .next()
+                                                                .and_then(|it| it.into_token())
+                                                                .map(|token| {
+                                                                    token
+                                                                        .text()
+                                                                        .trim()
+                                                                        .trim_start_matches('"')
+                                                                        .trim_end_matches('"')
+                                                                        .trim()
+                                                                        .to_string()
+                                                                })
                                                         });
```

Comment:
if you update to the latest `apollo-rs` commit you can get the STRING_VALUE as rust's `String` type:

```rust
if let ast::Value::StringValue(val) =
    argument.value().expect("Cannot get argument value.")
{
    let s: String = val.into();
}
```

It should help you eliminate this green node manipulation happening here.

Reply:
^ It is still preferable that you match on the variants of `argument.value()`, as not all variants implement casting to native rust types (they are not likely to in the future either).

Reply:
it works! 🙌 thank you! 🚀